### PR TITLE
Feature: Basic FileList Page

### PR DESCRIPTION
**Pull request by bohdan-meltiukhov**
# File List Page
This PR adds the FileList Page with nested component, such as:
- UserDetails,
- Breadcrumbs;
- File List;
- File Item.

### CODE REVIEW

File: ui-components/app/components/file-item/index.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      count.innerText = this._getReadableFileSizeString(this._size);
+    }
+
+    const cellActions = this.rootElement.querySelector('[data-test="cell-actions"]');
+    if (this._type === 'folder') {
+      cellActions.innerHTML = '<span class="glyphicon glyphicon-upload"></span>';
+    } else {
+      cellActions.innerHTML = '<span class="glyphicon glyphicon-download"></span>';
+    }
+    cellActions.innerHTML += ' <span class="glyphicon glyphicon-remove-circle"></span>';
```
Comment (dkalinin) [inline ui-components/app/components/file-item/index.js:116]: There's a lot of if-folder cases. Let's have 2 separate components for file and folder. 

- [ ] Refactor

File: ui-components/app/components/root-breadcrumbs/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import Component from '../component.js';
+
+/**
+ * The component for displaying breadcrumbs.
+ */
+export default class RootBreadcrumbs extends Component {
```
Comment (dkalinin) [inline ui-components/app/components/root-breadcrumbs/index.js:6]: I can't find usage of this component. What's the difference between `RootBreadcrumbs` and 

- [ ] Explain

File: ui-components/app/components/breadcrumbs/index.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    this._folder = folder;
+    this.render();
+  }
+
+  /** @inheritdoc */
+  markup() {
+    return `
+        <div data-test="breadcrumbs">
+            <div class="folder-icon">
+                <a href="#/file-list">
```
Comment (dkalinin) [inline ui-components/app/components/breadcrumbs/index.js:32]: Please use `FILE_LIST_ROUTE` here.

- [ ] Done

File: ui-components/app/components/file-item/index.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    let i = -1;
+    const byteUnits = [' KB', ' MB', ' GB', ' TB', 'PB', 'EB', 'ZB', 'YB'];
+    do {
+      fileSizeInBytes = fileSizeInBytes / 1024;
+      i++;
+    } while (fileSizeInBytes > 1024);
+
+    return Math.max(fileSizeInBytes, 0.1).toFixed(1) + byteUnits[i];
+  }
+}
```
Comment (dkalinin) [inline ui-components/app/components/file-item/index.js:87]: Seems that if `fileSizeInBytes` equals 0 (bytes), then result will be 0.1 KB. Let's show 0 B. 

- [ ] Done

File: ui-components/app/components/file-list/index.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  markup() {
+    return `
+        <table class="files" data-test="file-list-table"></table>
+    `;
+  }
+
+  /** @inheritdoc */
+  initNestedComponents() {
+    this._files.forEach((file) => {
+      const row = document.createElement('tr');
```
Comment (dkalinin) [inline ui-components/app/components/file-list/index.js:44]: Let's move `tr` to FolderItem/FileItem component.

- [ ] Done

File: ui-components/app/components/folder-item/index.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+   *
+   * @param {Element} container - The parent element for the folder item component.
+   * @param {Parameters} parameters - The initial folder items configurations.
+   */
+  constructor(container, parameters) {
+    super(container);
+
+    this._parameters = parameters;
+
+    console.log('name:' + this._parameters.name + '.');
```
Comment (dkalinin) [inline ui-components/app/components/folder-item/index.js:29]: Please remove.

- [ ] Done

File: ui-components/app/components/list-item/index.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    fakeElement.innerHTML = this.markup();
+
+    this.rootElement = fakeElement.firstElementChild;
+    const parentElement = this._container.parentElement;
+    parentElement.removeChild(this._container);
+    parentElement.appendChild(this.rootElement);
+
+    this.initNestedComponents();
+    this.addEventListeners();
+  }
```
Comment (dkalinin) [inline ui-components/app/components/list-item/index.js:21]: Let's not override `render()` method. 

- [ ] Done

File: ui-components/app/components/root-breadcrumbs/index.js (hunk 2 above)
Comment (bohdan-meltiukhov) [reply ui-components/app/components/root-breadcrumbs/index.js]: I removed it now.

File: ui-components/app/components/file-list/index.js (hunk 5 above)
Comment (bohdan-meltiukhov) [reply ui-components/app/components/file-list/index.js:44]: I still needed to create an element to pass to `FolderItem`s and `FileItem`s as a container. I changed it from `<tr>` to `<div>` now.

File: ui-components/app/components/list-item/index.js (hunk 7 above)
Comment (bohdan-meltiukhov) [reply ui-components/app/components/list-item/index.js:21]: Explained during the meeting.

File: ui-components/app/components/file-item/index.js [hunk 8]
```diff
@@ -82,6 +82,6 @@ export default class FileItem extends ListItem {
       i++;
     } while (fileSizeInBytes > 1024);
 
-    return Math.max(fileSizeInBytes, 0.1).toFixed(1) + byteUnits[i];
+    return fileSizeInBytes.toFixed(1) + byteUnits[i];
```
Comment (dkalinin) [inline ui-components/app/components/file-item/index.js:85]: - [ ] If file size is 0 B then show "0 B".
- [ ] If file size is 200 B then show "200 B".

File: ui-components/tests/components/folder-item/index.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+  const itemsCount = folderItem.querySelector('[data-test="cell-count"]');
+  assert.strictEqual(itemsCount.innerText, `${itemsNumber} items`, 'The folder item should display the provided ' +
+    'number of items.');
+
+  const actions = folderItem.querySelector('[data-test="cell-actions"]');
+  const firstAction = actions.firstElementChild;
+  assert.strictEqual(firstAction.className, 'glyphicon glyphicon-upload', 'The folder item should show the upload ' +
+    'action.');
+});
```
Comment (dkalinin) [inline ui-components/tests/components/folder-item/index.js:50]: - [ ] Let's have some tests to check that file size is displayed correctly.

**Pull request merged**